envoy/common/scope_tracker.h

```diff
@@ -18,6 +19,10 @@ class ScopeTrackedObject {
 public:
   virtual ~ScopeTrackedObject() = default;
 
+  /* If the tracked object has a ExecutionContext, this function can return a ScopedExecutionContext
+   * to active and deactive it. */
+  virtual ScopedExecutionContext scopedExecutionContext() const { return ScopedExecutionContext(); }
```

Comment:
`return {};` if the compiler allow. I think clang-tidy will complain this but it doesn't work for now (we will bring it back asap)

It would be better to change method to `ExecutionContext* executionContext() const { return nullptr; }`.

We should return `ExtensionContext*` here only and construct the `ScopedExecutionContext` in `ScopeTrackerScopeState`. By this way, we can make the `ScopedExecutionContext` non-moveable.